pdns/backends/gsql/gsqlbackend.cc

```diff
@@ -661,7 +663,22 @@ int GSQLBackend::addDomainKey(const DNSName& name, const KeyData& key)
   catch (SSqlException &e) {
     throw PDNSException("GSQLBackend unable to store key: "+e.txtReason());
   }
-  return 1; // XXX FIXME, no idea how to get the id
+
+  try {
+    d_GetLastInsertedKeyIdQuery_stmt->execute();
+    if (!d_GetLastInsertedKeyIdQuery_stmt->hasNextRow())
+      throw PDNSException("GSQLBackend unable to get id");
+    SSqlStatement::row_t row;
+    d_GetLastInsertedKeyIdQuery_stmt->nextRow(row);
+    int id = std::stoi(row[0]);
```

Comment:
missing ASSERT_ROW_COLUMNS() to prevent reading out of bound (custom queries !!!)